cmd/tuf/app/sign.go

```diff
@@ -176,10 +178,20 @@ func SignMeta(ctx context.Context, store tuf.LocalStore, name string, signer sig
 	}
 
 	// Get TUF public IDs associated to the signer.
-	pubKey, err := keys.ConstructTufKey(ctx, signer)
+	pubKey, err := keys.ConstructTufKey(ctx, signer, false)
```

Comment:
Just to check, does this need to be an `if else` for `addOldType`? Is it ok to have candidate key IDs for both the new and old format with only the old format being in use?

Reply:
The tuf clients should just ignore signatures from keys that does not count towards the signature threshold (i.e the wrong role for the key).

I have verified this for cosign by manually adding an unknown key-id in the signatures section.

I didn't go with a proper if/else to chose on using old, new or both as this is throw-away work after the signing, so adding both or just the new one was the path of least resistance.